nvtabular/tools/data_gen.py

```diff
@@ -0,0 +1,442 @@
+                # save vocab to file
+                ser.to_parquet(file_path)
+                col_vocab.append(file_path)
+            vocab_files.append(col_vocab)
+        return vocab_files
+
+    def merge_vocab(self, dataset_file_paths, vocab_file_paths, cats_rep, output):
+        # Load newly created dataset via Dataset
+        final_files = []
+        noe_df = Dataset(dataset_file_paths, engine="parquet")
+        # for each dataframe cycle through all cat col vocabs and left merge
+        for idx, df in enumerate(noe_df.to_iter()):
+            for x in range(0, len(cats_rep)):
+                # vocab_files is list of lists of file_paths
+                vocab_df = Dataset(vocab_file_paths[x])
+                for vdf in vocab_df.to_iter():
+                    col_name = cats_rep[x][0]
+                    focus_col = df[col_name]
+                    ser, offs = self.merge_cats_encoding(focus_col, vdf[col_name])
+                    if offs:
+                        df[col_name] = self.create_multihot_col(offs, ser)
+                    else:
+                        df[col_name] = ser
+            file_path = os.path.join(output, f"FINAL_{idx}.parquet")
+            df.to_parquet(file_path)
+            final_files.append(file_path)
+        return final_files
+
+    def verify_df(self, df_to_verify):
+        sts = []
+        ps = []
+        for col in df_to_verify.columns:
+            st_df, p_df = self.dist.verify(df_to_verify[col].to_pandas())
+            sts.append(st_df)
+            ps.append(p_df)
+        return sts, ps
+
+    def get_batch(self, row_size):
+        # grab max amount of gpu memory
+        gpu_mem = device_mem_size(kind="free") * self.gpu_frac
+        # find # of rows fit in gpu memory
+        return gpu_mem // row_size
+
+    def get_row_size(self, row, cats_reps):
+        """
+        row = cudf.DataFrame comprising of one row
+        """
+        size = 0
+        for col in row.columns:
+            if type(row[col].dtype) is cudf.core.dtypes.ListDtype:
+                # second from last position is max list length
+                # find correct cats_rep by scanning through all for column name
+                tar_rep = self.find_target_rep(col, cats_reps)
+                max_size = tar_rep[-2]
+                size = size + row[col]._column.elements.dtype.itemsize * max_size
+            else:
+                size = size + row[col].dtype.itemsize
+        return size
+
+    def find_target_rep(self, name, cats_rep):
+        for rep in cats_rep:
+            if name in rep[0]:
+                return rep
+
+
+distro_types = {"powerlaw": PowerLawDistro, "uniform": UniformDistro}
+
+
+class Col:
+    def __init__(self, name, dtype, distro=None):
+        self.name = name
+        self.dtype = dtype
+        self.distro = distro
+
+    def tupel(self):
+        tupel = []
+        for attr, val in self.__dict__.items():
+            tupel.append(val)
+        return tupel
```

Comment:
Whats this method trying to do - and what does 'tupel' mean here (is it supposed to be 'tuple' ?).  

I think this could be more cleanly expressed like :
```suggestion
    def tupel(self):
        return list(self.__dict__.values())
```